pkg/controller/directvolumemigration/rsync.go

```diff
@@ -159,16 +159,77 @@ func (t *Task) getRsyncClientMutations(client compat.Client, namespace string) (
 	if err != nil {
 		return nil, liberr.Wrap(err)
 	}
-	runAsUser := int64(0)
-	trueBool := bool(true)
-	customSecurityContext := &corev1.SecurityContext{
-		Privileged:             &isPrivileged,
-		RunAsUser:              &runAsUser,
-		ReadOnlyRootFilesystem: &trueBool,
-		Capabilities: &corev1.Capabilities{
-			Drop: []corev1.Capability{"MKNOD", "SETPCAP"},
-		},
+
+	destClient, err := t.getDestinationClient()
+	if err != nil {
+		return nil, liberr.Wrap(err)
+	}
+
+	migration, err := t.Owner.GetMigrationForDVM(destClient)
+	if err != nil {
+		return nil, liberr.Wrap(err)
+	}
+	customSecurityContext := &corev1.SecurityContext{}
+	if migration.Spec.RunAsRoot != nil && *migration.Spec.RunAsRoot {
+		// if rsync needs to run as root
+		checkLabels, err := checkUserNamespaceLabel(client)
+		if err != nil {
+			return nil, liberr.Wrap(err)
+		}
+		privilegedLabelPresent := true
```

Comment:
nit: doesn't matter in this case, but default should be `false`

Reply:
I have set it `true` because in the case where we don't need the labels present in the namespace to run as root (< 4.12+), we don't check for labels and hence it's always `true`. In the event of the missing labels when we need them we set the variable to `false`.